app/src/test/java/org/mozilla/fenix/gleanplumb/state/MessagingMiddlewareTest.kt

```diff
@@ -245,42 +245,63 @@ class MessagingMiddlewareTest {
     }
 
     @Test
-    fun `WHEN updateMessage THEN update available messages`() = runTestOnMain {
+    fun `WHEN updateMessage THEN update available messages in same order`() = runTestOnMain {
         val oldMessage = Message(
             "oldMessage",
-            MessageData(),
+            MessageData(surface = MessageSurfaceId.HOMESCREEN),
             action = "action",
             mockk(relaxed = true),
             listOf("trigger"),
             Message.Metadata("same-id", pressed = false),
         )
 
+        val secondOldMessage = Message(
+            "secondOldMessage",
+            mockk(relaxed = true),
+            action = "action",
+            mockk(relaxed = true),
+            listOf("trigger"),
+            Message.Metadata("different-id", pressed = false),
+        )
+
         val updatedMessage = Message(
             "oldMessage",
-            MessageData(),
+            MessageData(surface = MessageSurfaceId.HOMESCREEN),
             action = "action",
             mockk(relaxed = true),
             listOf("trigger"),
             Message.Metadata("same-id", pressed = true),
         )
 
+        val secondUpdatedMessage = Message(
+            "secondOldMessage",
+            mockk(relaxed = true),
+            action = "action",
+            mockk(relaxed = true),
+            listOf("trigger"),
+            Message.Metadata("different-id", pressed = true),
+        )
+
         val spiedMiddleware = spyk(middleware)
 
         val appState: AppState = mockk(relaxed = true)
         val messagingState: MessagingState = mockk(relaxed = true)
 
+        every { messagingState.messages } returns listOf(oldMessage, secondOldMessage)
```

Comment:
I realize you didn't write this test, but it doesn't actually test anything since we're returning the data we are asserting.

Let's write a separate test instead that uses a real store so we can test that the middleware is not changing the message position. We should also file an issue to refactor these tests to use a real store and not a mock.